core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -501,15 +502,35 @@ public static String cleanPath(String path) {
     }
 
     public static Set<String> splitStringByCommaToSet(final String s) {
-        return splitStringToSet(s, ',');
+        return splitStringToSet(s, ',', false);
     }
 
     public static String[] splitStringByCommaToArray(final String s) {
         if (s == null || s.isEmpty()) return Strings.EMPTY_ARRAY;
         else return s.split(",");
     }
 
-    public static Set<String> splitStringToSet(final String s, final char c) {
+    public static Set<String> splitStringByCommaAndTrim(final String s) {
```

Comment:
yeah i wasn't sure what was best, most uses are reading rest request params which shouldn't have the whitespace anyway.  i'm up for changing it to always do trim (unless that's the delimiter char) and not have that option on the method.  that would make my extra wrapper method even more unnecessary and could go away.